server/server.js

```diff
@@ -1,15 +1,92 @@
+const dotenv = require("dotenv");
+dotenv.config();
 const express = require("express");
+const cors = require("cors");
 const app = express();
 const port = process.env.PORT || 5000;
+const database = require("./dbConfig");
 
-app.listen(port, () => console.log(`Listening on port ${port}`));
+app.use(express.json());
+app.use(
+  cors({
+    origin: "*",
+  })
+);
+
+// GET "/videos"
+app.get("/videos", async function (request, response) {
+  try {
+    const order = request.query.order || "desc";
+    const sqlQuery = `SELECT * FROM videos ORDER BY rating ${order};`;
+    const getAllVideos = await database.query(sqlQuery);
+    console.log(getAllVideos.rows);
+    const allVideos = getAllVideos.rows;
+    return response.status(200).json(allVideos);
+  } catch (error) {
+    return response.status(500).json({ error });
+  }
+});
+
+app.get("/videos/:id", async function (request, response) {
+  try {
+    const videoId = request.params.id;
+    const getVideoById = await database.query(
+      "SELECT * FROM videos WHERE id = $1",
+      [videoId]
+    );
+
+    if (getVideoById.rows.length === 0) {
+      return response
+        .status(400)
+        .json({ error: `The video with id ${videoId} doesn't exist` });
```

Comment:
Nice specific error messages, great idea to tell the client exactly what went wrong!

Reply:
Thank you Sam.